### ISSUE TITLE
Logo?

### ISSUE THREAD

**Issue by alexispurslane** [matz/streem#57]
I was thinking a png icon along the lines of:
![streem](https://cloud.githubusercontent.com/assets/1920151/5550888/23eeb640-8b67-11e4-90ed-6d78556db1d5.png)

BTW, I'm not really that great at drawing, so any other icons would be great!

**Comment by halogenandtoast** [comment]
![streem-logo](https://cloud.githubusercontent.com/assets/201288/5729229/dcd48afe-9b3e-11e4-8efe-f89d2442167d.png)

My take on a logo.

**Comment by nicolasmccurdy** [comment]
That looks neat, though I think it would prefer it if the black bars on the length had more equal lengths.

**Comment by christopherdumas** [comment]
I like it. Especially since Matz decided to stay with plain pipes. Also, it has a cool Asian-ish-ness.

**Comment by shunsukeaida** [comment]
Maybe you could merge <samp>川</samp> and <samp>m</samp>;

**Comment by markburns** [comment]
FYI anyone that doesn't know 川 means stream or river. I like the idea,
perhaps there is a font though that is more regular in size for the 3 bars.

**Comment by halogenandtoast** [comment]
When I get a chance I'll fix the height of the bars and some of the spacing problems I have with it. It was a quick mock-up while I was at work.

**Comment by ferdinandrosario** [comment]
@nicolasmccurdy +1 for your logo

**Comment by halogenandtoast** [comment]
Revised the logo a bit, changed the pipe heights and fixed some spacing/sizing issues

![streem-revised](https://cloud.githubusercontent.com/assets/201288/5777376/03082b72-9d60-11e4-9d2f-8566b35a1c15.png)

**Comment by nicolasmccurdy** [comment]
Would having equal pipe lengths look weird for being confusing to someone who reads Japanese? We could also try using a different typeface for the Japanese character.

**Comment by terlar** [comment]
Also +1 on what @shunsukeaida suggests. Would be intresting to see what the 川 character looked like as an m replacement.

Like: stree川

**Comment by halogenandtoast** [comment]
Some variations (I don't think I like 川 as the m)
![streem-m](https://cloud.githubusercontent.com/assets/201288/5783820/9e92ff7a-9d94-11e4-81c4-0bf04eb837b4.png)
![streem-m-color-2](https://cloud.githubusercontent.com/assets/201288/5783821/a1852a46-9d94-11e4-993c-51f4190dd8ad.png)

**Comment by halogenandtoast** [comment]
With a smaller 川 it's slightly better

![streem-smaller-m](https://cloud.githubusercontent.com/assets/201288/5783874/1d856714-9d95-11e4-8f35-cce8e62421e1.jpg)

**Comment by nicolasmccurdy** [comment]
I think replacing the m with 川 is way too confusing, it's not obvious that it's supposed to be m. We could try adding lines connecting them on top, but I think that defaults the purpose of it. I prefer the logo on the left.

**Comment by christopherdumas** [comment]
I agree. The logo with the three bars on the left is better.

**Comment by nicolasmccurdy** [comment]
@halogenandtoast I'm starting to really like the idea of having a 川 logo, and I want to try it out with my design. What font did you use?

**Comment by nicolasmccurdy** [comment]
I just realized that the logo font looks like Hiragino Kaku Gothic Pro W6.

**Comment by nicolasmccurdy** [comment]
How about something like this?

![Streem logo](http://i.imgur.com/o6AHwe1.png)

**Comment by shunsukeaida** [comment]
Maybe it depends on each cultural roots, but for me, 川 is more than just three bars. It can easily read as Kawa-Streem, so we should carefully separate it from the rest of chars (that is "Streem"). In short, we shouldn't make it recognizable as a Kanji char at the first glance. 

1) Puts it into behind of "Streem"
2) Use a different font, color, opacity than "Streem"
3) Merge 川 with m

3) was my original suggestion, and I either didn't think replacing m by 川 would work. Rather, a 川 with the arches between bars. 

I'm terrible at photoshoping, so sorry for no actual sample offering.

**Comment by markburns** [comment]
I agree with @shunsukeaida but also can't offer any samples. 

I think the typography of the word itself on @nicolasmccurdy 's is better. This one:

![image](https://cloud.githubusercontent.com/assets/104158/5788027/cf20ff9a-9e1e-11e4-80ff-29f014370a0b.png)

I like the idea of incorporating 川, I think it looks confusing replacing the m, and agree that it makes it read like kawa streem.

I think a slightly altered version of 川　(the ones here look more like serif fonts) that fits the feel of the font of @nicolasmccurdy 's contribution and merges into the m. I'd like to see it with the same color, but also with contrasting color/opacity etc to see if it stands out and reads in a nice dual clever way, rather than a slightly contrived bolted-on bit of Japanese-ness.

**Comment by matz** [comment]
I am sorry but I am not going to use the character 川 in the logo.  Maybe it appears cool for non Asian people, but it hurts my eyes.

**Comment by nicolasmccurdy** [comment]
Alright. Should we just use a `|` (pipe) for the icon, or do we have any other ideas? We could always not have an icon and just use "Streem", possibly with "a stream based concurrent scripting language" as the subheading.

**Comment by nicolasmccurdy** [comment]
@matz Also, do you think the issue with 川 could be remedied with a different Japanese font?

**Comment by matz** [comment]
@nicolasmccurdy I wouldn't agree if 川 is in predefined font. If it's designed in like carigraphy, it might be acceptable. For the time being, I prefer just "Streem" maybe with "|" (pipe).

**Comment by nicolasmccurdy** [comment]
How about these?

![Streem logo with one bar](http://i.imgur.com/65UIalP.png)
![Streem logo with two bars](http://i.imgur.com/JebcDVq.png)

**Comment by christopherdumas** [comment]
yeah, that works. I like the second one best.

**Comment by nicolasmccurdy** [comment]
I agree, I think the second one looks more balanced and fits the theme of the language better.

**Comment by tech4david** [comment]
Here's what I've created, heavily based on @nicolasmccurdy's logo. I've made the pipes the same height as the text though, I think it looks nicer despite it not being how it would actually look. Anyone who wants the svg, just ask
![streem logo](https://cloud.githubusercontent.com/assets/4248177/7833074/c13c8086-045b-11e5-84e1-2230c9d0e086.png)

**Comment by tech4david** [comment]
Update, I forgot to middle-align the left pipe properly
![streem logo](https://cloud.githubusercontent.com/assets/4248177/7833263/6842c06a-045d-11e5-9e5b-7300df7a2549.png)

**Comment by tech4david** [comment]
Here is a version with no empty space around the image, i.e. like the ones by @nicolasmccurdy 
![streem logo](https://cloud.githubusercontent.com/assets/4248177/7833439/cca80ca8-045e-11e5-98f2-9588c42c9d4e.png)

**Comment by nicolasmccurdy** [comment]
@tech4david :+1:

**Comment by tech4david** [comment]
@nicolasmccurdy glad you like it (even though I kinda just stole yours :D sorry)
@matz any suggestions?

**Comment by dragostis** [comment]
![streem 1-01](https://cloud.githubusercontent.com/assets/4136413/8145099/eea785e8-1202-11e5-98c6-e4d8742e99f0.png)

**Comment by dragostis** [comment]
Most programming languages are very easy to recognize due to their colors, so streem should use colors that set it apart.

I chose an elegant grotesque font and used nice color blend.

![streem_logo](https://cloud.githubusercontent.com/assets/4136413/8148335/1d25a942-12a1-11e5-9623-a3743dfab7b3.png)
![streem_symbol](https://cloud.githubusercontent.com/assets/4136413/8148337/24af7530-12a1-11e5-9a94-01436678e281.png)
![streem_colors](https://cloud.githubusercontent.com/assets/4136413/8148338/29c4130a-12a1-11e5-9afb-df70480994d5.png)
![streem_negative](https://cloud.githubusercontent.com/assets/4136413/8148339/2e49c802-12a1-11e5-8a48-b30748c0c4fb.png)
![streem_alternative](https://cloud.githubusercontent.com/assets/4136413/8148340/323d5046-12a1-11e5-9e59-561f6910523f.png)

**Comment by naturalethic** [comment]
@dragostis I like where you're going with this.  Best submissions thus far.

**Comment by tech4david** [comment]
[This one] is nice but IMO the other variations (last two particularly) are too dull - need more colours. Also:
> @matz
> For the time being, I prefer just "Streem" maybe with "|" (pipe).

Maybe make another version without the `>`? Nice work though

[This one]: https://cloud.githubusercontent.com/assets/4136413/8148335/1d25a942-12a1-11e5-9623-a3743dfab7b3.png

**Comment by dragostis** [comment]
The alternatives are for different use cases. Here's another version.

![streem_logo](https://cloud.githubusercontent.com/assets/4136413/8150216/63c0a832-12e9-11e5-806c-d881890f05e8.png)

**Comment by dimanyc** [comment]
@dragostis love it

**Comment by karmaQ** [comment]
Hi @matz , I really like Ruby and Streem, and asked my girlfriend to design a logo fews months ago. Now, here it is.

![streem-logo](https://cloud.githubusercontent.com/assets/1725827/10409539/46f0890a-6f57-11e5-90f2-cef493a7f08e.png)

**Comment by ferdinandrosario** [comment]
@karmaQ Super :clap: :clap: :clap: :yellow_heart: :green_heart: :heart: :star: :star2:

**Comment by davydovanton** [comment]
great work @karmaQ and @dragostis :clap:

**Comment by franzejr** [comment]
@karmaQ Liked!

**Comment by matz** [comment]
Wow, it's so professional!

**Comment by jonathanmarvens** [comment]
Wow. Tell your girlfriend she did a really great job, @karmaQ!

**Comment by karmaQ** [comment]
@jonathanmarvens , I told her :
<img width="720" alt="1" src="https://cloud.githubusercontent.com/assets/1725827/10506209/d7f22762-7347-11e5-837b-0de508f7d80c.png">

she was so happy and rewarding, thank you.

**Comment by karmaQ** [comment]
Everyone, thanks!
Here is a new one and another is being creative. Which do you like.

![streem-](https://cloud.githubusercontent.com/assets/1725827/10506309/e1e39598-7348-11e5-863b-c005fff27982.png)

**Comment by ferdinandrosario** [comment]
this looks similar to http://elixir-lang.org/ logo. this is not  a negative feedback .... appreciated your work a lot........ :heart: :+1:

**Comment by karmaQ** [comment]
@ferdinandrosario Yes, similar to elixir's logo, although I don't  think she was know elixir. Another is  coming soon, please wait ~

**Comment by ferdinandrosario** [comment]
:clap:

**Comment by phantomk** [comment]
@karmaQ  i like this one

**Comment by shunsukeaida** [comment]
![e1e39598-7348-11e5-863b-c005fff27982](https://cloud.githubusercontent.com/assets/5036100/10513878/796beca8-7385-11e5-9f2c-93d8e8179699.png)

This one is actually my favorite  (despite being a prototype), since a logo of streem should be something dynamic, and others look more like a static drop.

**Comment by dragostis** [comment]
@karmaQ This is a really cool and creative design! I love it! The idea behind it is really sweet.

One thing that I was considering when I did my sketch was to try and use a palette that wasn't used before when choosing the design philosophy behind a programming language. Since GitHub uses colors to differentiate programming languages (ruby - red, Python - blue, Scala - turquoise, JavaScript - yellow), I think Streem can benefit greatly from a differentiating palette which would help consolidate it in the user's minds. Since I really like the current idea, I would suggest trying a few sketches with a different palette and maybe with a typeface added for the logo. ([Brandon Grotesque](https://www.myfonts.com/fonts/hvdfonts/brandon-grotesque/) can work here too, but I'm sure there's a better alternative.)

**Comment by h1u2i3** [comment]
May be we can start from https://en.wikipedia.org/wiki/Yin_and_yang? Some of the functions of this language are similar to those expressed by Yin and Yang

**Comment by terry-xiaoyu** [comment]
Great design from Yiling. I like the 'S'-like one most 👍
